### PR TITLE
Fix aligner errormessages

### LINKED ISSUE #1809: errors are not properly logged in analysis.align
### Expected behaviour
If analysis.align.AlignTrj or get_matching_atoms() fails to find a mobile and ref atom group that correspond to each other then they should print/log all diagnostic messages.

### Actual behaviour
Only part of the messages were printed or logged.

### Code to reproduce the behaviour

You can use the gist https://gist.github.com/orbeckst/2686badcd15031e6c946baf9164a683d to play with errors. 

It also shows that the current get_matching_atoms() is really not very smart and could be improved substantially to automatically find good matches.

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.17.1-develop

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #1809

Changes made in this Pull Request:
 - properly output error messages in analysis.align
 - catches some edge cases that now raise SelectionError
 - added tests for these cases (but not for the output)

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by orbeckst** [comment]
@MDAnalysis/coredevs could someone please review? Thanks.

(Just rebased against develop so that it can be merged when the tests and reviews pass.)

File: package/MDAnalysis/analysis/align.py [hunk 1]
```diff
@@ -1028,34 +1028,29 @@ def get_matching_atoms(ag1, ag2, tol_mass=0.1, strict=False):
...
             msg = ("Reference and trajectory atom selections do not contain "
                    "the same number of atoms: \n"
                    "atoms:    N_ref={0}, N_traj={1}").format(
-                ag1.n_atoms, ag2.n_atoms)
+                       ag1.n_atoms, ag2.n_atoms)
             if strict:
+                logger.error(msg)
                 raise SelectionError(msg)
 
             # continue with trying to creating a valid selection
```
Comment (tylerjereddy) [inline package/MDAnalysis/analysis/align.py]: `creating` -> `create` typo, but shouldn't kill the messenger for cleaning up other parts of the module

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 2]
```diff
@@ -71,6 +71,64 @@ def test_exception(self):
...
+        ref = reference.select_atoms(selection)
+        mobile = universe.select_atoms(selection)
+        groups = align.get_matching_atoms(ref, mobile, strict=strict)
+        assert_equal(groups[0].names, groups[1].names)
+
+    @pytest.mark.parametrize("strict", (True, False))
+    def test_nomatch_atoms_raise(self, universe, reference,
+                                 strict, selection="protein and backbone"):
+        # one atom less but same residues; with strict=False should try
+        # to get selections (but current code fails, so we also raise SelectionErro)
```
Comment (tylerjereddy) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: `SelectionErro` missing the final `r`?

File: package/MDAnalysis/analysis/align.py (hunk 1 above)
Comment (orbeckst) [reply package/MDAnalysis/analysis/align.py]: fixed

File: testsuite/MDAnalysisTests/analysis/test_align.py (hunk 2 above)
Comment (orbeckst) [reply testsuite/MDAnalysisTests/analysis/test_align.py]: fixed

**Comment by orbeckst** [comment]
Thanks @tylerjereddy . I fixed the spelling, rebased and force pushed.

**Pull request merged**